modules/javafx.controls/src/test/java/test/javafx/scene/control/ContextMenuTest.java

```diff
@@ -636,4 +640,106 @@ private ContextMenu createContextMenuAndShowSubMenu() {
         assertEquals("Expected " + item1.getText() + ", found " + focusedItem.getText(),
                 item1, focusedItem);
     }
+
+    @Test public void test_position_showOnScreen() {
+        ContextMenu cm = createContextMenu(false);
+        cm.show(anchorBtn,100, 100);
+
+        assertEquals(cm.getAnchorX(), 100, 0.0);
+        assertEquals(cm.getAnchorY(), 100, 0.0);
+    }
+
+    @Test public void test_position_showOnTop() throws InterruptedException {
+        ContextMenu cm = createContextMenu(false);
+        cm.show(anchorBtn, Side.TOP, 0, 0);
+
+        Bounds anchorBounds = anchorBtn.localToScreen(anchorBtn.getLayoutBounds());
+        Node cmNode = cm.getScene().getRoot();
+        Bounds cmBounds = cm.getScene().getRoot().localToScreen(cmNode.getLayoutBounds());
+
+        assertEquals(anchorBounds.getMinX(), cmBounds.getMinX(), 0.0);
+        assertEquals(anchorBounds.getMinY(), cmBounds.getMaxY(), 0.0);
+    }
+
+    @Test public void test_position_showOnTopOffset() throws InterruptedException {
+        ContextMenu cm = createContextMenu(false);
+        cm.show(anchorBtn, Side.TOP, 3, 5);
+
+        Bounds anchorBounds = anchorBtn.localToScreen(anchorBtn.getLayoutBounds());
+        Node cmNode = cm.getScene().getRoot();
+        Bounds cmBounds = cm.getScene().getRoot().localToScreen(cmNode.getLayoutBounds());
+
+        assertEquals(anchorBounds.getMinX() + 3, cmBounds.getMinX(), 0.0);
+        assertEquals(anchorBounds.getMinY() + 5, cmBounds.getMaxY(), 0.0);
+    }
+
+    @Test public void test_position_withCSS() throws InterruptedException {
+        anchorBtn.getScene().getStylesheets().add(createStylesheet());
+        test_position_showOnTop();
+        test_position_showOnRight();
+        test_position_showOnLeft();
+        test_position_showOnBottom();
+    }
+
+    private String createStylesheet() {
+        try {
+            File f = File.createTempFile("test_position_showOnTopWithCSS", ".css");
```

Comment:
There is no need to create a temporary file for this css resource, since its contents are fixed. Instead, you should add `test_position_showOnTopWithCSS.css` to the repo under [`modules/javafx.controls/src/test/resources/test/javafx/scene/control/`](https://github.com/openjdk/jfx/tree/master/modules/javafx.controls/src/test/resources/test/javafx/scene/control). Then you can just do this:

```
    return ContextMenuTest.class.getResource("test_position_showOnTopWithCSS.css").toExternalForm();
```

Reply:
Thanks for pointing this out. I took the example from the bug entry (where it was nice to have everything in one file). But in the test it is of course much more elegant to use a .css file from the resources.
Adapted the test accordingly.